plugin.video.vtm.go/resources/lib/kodiwrapper.py

```diff
@@ -0,0 +1,583 @@
+            listing.append((url, list_item, is_folder))
+
+        succeeded = xbmcplugin.addDirectoryItems(self._handle, listing, len(listing))
+        xbmcplugin.endOfDirectory(self._handle, succeeded, cacheToDisc=cache)
+
+    @staticmethod
+    def _generate_listitem(title_item):
+        """ Generate a ListItem from a TitleItem """
+        from xbmcgui import ListItem
+
+        # Three options:
+        #  - item is a virtual directory/folder (not playable, path)
+        #  - item is a playable file (playable, path)
+        #  - item is non-actionable item (not playable, no path)
+        is_folder = bool(not title_item.is_playable and title_item.path)
+        is_playable = bool(title_item.is_playable and title_item.path)
+
+        list_item = ListItem(label=title_item.title, path=title_item.path)
+
+        if title_item.prop_dict:
+            list_item.setProperties(title_item.prop_dict)
+        list_item.setProperty(key='IsPlayable', value='true' if is_playable else 'false')
+
+        list_item.setIsFolder(is_folder)
+
+        if title_item.art_dict:
+            list_item.setArt(title_item.art_dict)
+
+        if title_item.info_dict:
+            # type is one of: video, music, pictures, game
+            list_item.setInfo(type='video', infoLabels=title_item.info_dict)
+
+        if title_item.stream_dict:
+            # type is one of: video, audio, subtitle
+            list_item.addStreamInfo('video', title_item.stream_dict)
+
+        if title_item.context_menu:
+            list_item.addContextMenuItems(title_item.context_menu)
+
+        return list_item
+
+    def play(self, title_item, license_key=None):
+        """ Play the passed TitleItem.
+        :type title_item: TitleItem
+        :type license_key: string
+        """
+        play_item = self._generate_listitem(title_item)
+        play_item.setProperty('inputstreamaddon', 'inputstream.adaptive')
+        play_item.setProperty('inputstream.adaptive.max_bandwidth', str(self.get_max_bandwidth() * 1000))
+        play_item.setProperty('network.bandwidth', str(self.get_max_bandwidth() * 1000))
+        play_item.setProperty('inputstream.adaptive.manifest_type', 'mpd')
+        play_item.setMimeType('application/dash+xml')
+        play_item.setContentLookup(False)
+
+        if license_key is not None:
+            play_item.setProperty('inputstream.adaptive.license_type', 'com.widevine.alpha')
+            play_item.setProperty('inputstream.adaptive.license_key', license_key)
+
+        if title_item.subtitles_path:
+            play_item.setSubtitles(title_item.subtitles_path)
+
+        # To support video playback directly from RunPlugin() we need to use xbmc.Player().play instead of
+        # setResolvedUrl that only works with PlayMedia() or with internal playable menu items
+        xbmcplugin.setResolvedUrl(self._handle, True, listitem=play_item)
+
+        if self.get_setting_as_bool('showsubtitles'):
```

Comment:
This won't work: if a stream is already playing (video or audio), the `xbmc.Player().isPlaying()` will be `True` and the subtitles will be show on that stream, not the one that will start playing. I had the exact same issue in Retrospect. You will also need to check that `xbmc.Player().isPlaying()` and that it is playing your requested url.

Reply:
Hmm, this makes sense. This code is based on how the VRT NU addon does this, maybe they also have the same issue then. I'll update this!